qulice-pmd/src/main/java/com/qulice/pmd/rules/ProhibitOldStyleAssertionsRule.java

```diff
@@ -0,0 +1,198 @@
+ * @version $Id$
+ */
+public final class ProhibitOldStyleAssertionsRule extends AbstractJUnitRule {
+
+    /**
+     * Mask of prohibited imports.
+     */
+    private static final String[] PROHIBITED = {
+        "org.junit.Assert.assert",
+        "junit.framework.Assert.assert",
+    };
+
+    @Override
+    public Object visit(final ASTMethodDeclaration method, final Object data) {
+        if (this.isJUnitMethod(method, data)
+            && this.containsOldStyleAssert(method.getBlock())) {
+            this.addViolation(data, method);
+        }
+        return data;
+    }
+
+    @Override
+    public Object visit(final ASTImportDeclaration imp, final Object data) {
+        if (ProhibitOldStyleAssertionsRule.isArrayMatchesValue(
+            imp.getImportedName(), ProhibitOldStyleAssertionsRule.PROHIBITED
+        )) {
+            this.addViolation(data, imp);
+        }
+        return super.visit(imp, data);
+    }
+
+    /**
+     * Checks if string matches at least one element in array.
+     * @param value String value to be matched
+     * @param array Array of templates
+     * @return False if no matches found, true if at least one
+     */
+    private static boolean isArrayMatchesValue(final String value,
+        final String... array) {
+        boolean matches = false;
+        for (final String element : array) {
+            if (value.contains(element)) {
+                matches = true;
+                break;
+            }
+        }
+        return matches;
+    }
+
+    /**
+     * Recursively verifies if node contains old style assert statements.
+     * @param node Root statement node to search
+     * @return True if statement contains old style assertions, false otherwise
+     */
+    private boolean containsOldStyleAssert(
+        final Node node
+    ) {
+        boolean found = false;
+        if (node instanceof ASTStatementExpression
+            && ProhibitOldStyleAssertionsRule.isAssertStatement(node)) {
+            found = true;
+        }
+        if (!found) {
+            for (int iter = 0; iter < node.jjtGetNumChildren(); iter += 1) {
+                final Node child = node.jjtGetChild(iter);
+                if (this.containsOldStyleAssert(child)) {
+                    found = true;
+                    break;
+                }
+            }
+        }
+        return found;
+    }
+
+    /**
+     * Tells if the statement is an assert statement or not.
+     * @param statement Root node to search assert statements
+     * @return True is statement is assert, false otherwise
+     */
+    private static boolean isAssertStatement(final Node statement) {
+        boolean assrt;
+        try {
+            final ASTPrimaryExpression expression =
+                ProhibitOldStyleAssertionsRule.extractExpressionFrom(statement);
+            final ASTPrimaryPrefix prefix =
+                ProhibitOldStyleAssertionsRule.extractPrefixFrom(expression);
+            final String img =
```

Comment:
@vkuchyn Since we're using all these methods exactly once, let's just inline all the code in `isAssertStatement`. Currently, the code is jumping around just chaining one return value to the other, and it's actually not very readable.